recipes-omnect/initrdscripts/omnect-os-initramfs/common-sh

```diff
@@ -312,13 +313,21 @@ flashmode_efi_grub_handling()
 
     if [[ " ${machine_features} " =~ " efi " ]]; then
         echo "Configuring efi"
-        # delete all efi entries but last
-        local efi_entries=$(efibootmgr | grep ^Boot[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]\* | awk '{print $1}'| sed  '$ d')
+        # delete all efi entries
+        local efi_entries=$(efibootmgr | grep ^Boot[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]\* | awk '{print $1}')
         for i in ${efi_entries}; do
             i=${i#Boot}
             i=${i%\*}
             run_cmd_stdout efibootmgr --bootnum ${i} --delete-bootnum || return 1
         done
-        run_cmd_stdout efibootmgr --create --disk ${dest_dev} --part 1 --label "omnect_os" --loader \\\\EFI\\\\BOOT\\\\bootx64.efi || return 1
+
+        run_cmd_stdout mkdir -p /tmp/boot
+        run_cmd_stdout mount ${dest_dev}${p}1 /tmp/boot
+        run_cmd_stdout efibootmgr --create --disk ${dest_dev} --part 1 --label "omnect_os" --loader \\\\EFI\\\\BOOT\\\\bootx64.efi
+        # same entry with another label ... for debug purposes, when booting after flash-mode-{1,2} fails
+        run_cmd_stdout efibootmgr --create --disk ${dest_dev} --part 1 --label "omnect_os " --loader \\\\EFI\\\\BOOT\\\\bootx64.efi
+        efibootmgr -v > /tmp/boot/EFI/BOOT/efibootmgr_entry
```

Comment:
out of curiosity: is that for informational/debugging purposes only?

Reply:
correct